extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -480,6 +513,146 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
+   * resolves to the time that the RTC callout took to complete.
+   * If disableStalewhilerevalidate == true, then we will only
+   * send one RTC request per page. If it's !== true, then we always
+   * send two requests. The first will try
+   * to hit browser cache, and the second request will always bypass.
+   * This is a way for us to implement a simple stale-while-revalidate
+   * model.
+   * The targeting info from the RTC updates the targeting info on
+   * this object within mergeRtc.
+   * @return {?Promise<?number|?string>} The time the RTC callout took
+   *   (if successful) or an error message if rejecting.
+   * @private
+   */
+  executeRtc_() {
+    if (rtcPromise) {
+      return this.mergeRtc();
+    }
+    const ampRtcPageElement = document.getElementById('amp-rtc');
+    if (!ampRtcPageElement) {
+      return Promise.resolve();
+    }
+    let endpoint;
+    rtcConfig = tryParseJson(ampRtcPageElement.textContent);
+    if (!isObject(rtcConfig) || !(endpoint = rtcConfig['endpoint']) ||
+        typeof endpoint != 'string' || !isSecureUrl(endpoint)) {
+      user().warn(TAG,
+          `invalid RTC config: ${ampRtcPageElement.textContent}`);
+      return Promise.resolve();
+    }
+    let rtcTotalTime;
+    const disableSWR = (
+        rtcConfig['disableStaleWhileRevalidate'] === true);
+    const startTime = Date.now();
+    // Because we are wrapping the RTC request in the timeout,
+    // we are guaranteeing that if the RTC is slow to return and
+    // times out for the first slot, it won't be used for any
+    // other slots either. This is in opposition to the other way
+    // we could potentially do it, in which we only still make one
+    // rtc callout, but there is a 1 second timeout from the
+    // perspective of each slot. I.e. if the top slot on the page
+    // calls out for the RTC, which returns after 5 seconds, and
+    // then a slot way down on the page asks for it later.
+    rtcPromise = timerFor(window).timeoutPromise(
+        RTC_TIMEOUT,
+        xhrFor(this.win).fetchJson(
+            endpoint, {credentials: 'include'}).then(res => {
+              rtcTotalTime = Date.now() - startTime;
+              if (!disableSWR) {
+                // Repopulate the cache.
+                const headers = new Headers();
+                headers.append('Cache-Control', 'max-age=0');
+                xhrFor(this.win).fetchJson(endpoint, {
+                  credentials: 'include',
+                  headers,
+                }).catch(err => {
+                  user().error(TAG, err.message);
+                });
+              }
+              // Non-200 status codes are forbidden for RTC.
+              // TODO: Add to fetchResponse the ability to
+              // check for redirects as well.
+              if (res.status != 200) {
+                return null;
+              }
+              return res.text().then(text => {
+                if (text == '') {
```

Comment:
Add a comment indicating that empty is assumed to indicate ok response with nothing to merge.  Can also just do if (!text)